### PR TITLE
Fix translation of RWTexture subscript operations for Vulkan

### LINKED ISSUE #615: RWTexture2D doesn't work in Compute.hlsl
In latest slang ComputeShader sample doesn't work on ubuntu anymore. It seems to be problem when writing into RWTexture2D. 
The error output is listed below:

ERROR: /home/ANT.AMAZON.COM/chenka/Falcor3IntegrationWS/src/Falcor/.dex_project/compute.hlsl:66: '[]' : scalar integer expression required 
ERROR: /home/ANT.AMAZON.COM/chenka/Falcor3IntegrationWS/src/Falcor/.dex_project/compute.hlsl:66: 'gOutput_0' :  left of '[' is not of type array, matrix, or vector  
ERROR: /home/ANT.AMAZON.COM/chenka/Falcor3IntegrationWS/src/Falcor/.dex_project/compute.hlsl:66: '' : compilation terminated 
ERROR: 3 compilation errors.  No code generated.

Can this be fixed? thanks.

### PR DESCRIPTION
**Pull request by tfoleyNV**
Partially fixes #615

There's kind of a mess going on here, and it is difficult to be sure which of the changes here are strictly necessary.
Also, our testing isn't setup to run tests that use `RWTexture2D`, so the only testing I can really run is manual tests using Falcor.

The most basic issue here is that in an earlier change I added `ref` accessors for the subscript operation on various `RW*` types in the standard library, and that included `RWTexture2D` (and the other `RWTexture*` types). The compiler ended up favoring a `ref` accessor over a `set` accessor even when the `set` would suffice, but only the `set` accessor could be lowerd to GLSL/SPIR-V.

This change ends up implementing two different fixes for the same problem:

* Logic has been added to try and favor a `set` accessor over a `ref` accessor in the cases where either could be used (but still require a `ref` accessor to be used when it is really needed)

* The `ref` accessor for `RWTexture*` has been removed, since it turns out that the operations that might have benefited from it (atomics, and component-granularity stores) aren't actually allowed on typed UAVs anyway.

There is a deeper issue here that somebody needs to go through and rationalize our representation and handling of accessors like this, but I'm not going to be able to do that in the time I can put into this PR.

### CODE REVIEW

**Comment by KaChen3000** [comment]
thanks for quick fix.

**Comment by tfoleyNV** [comment]
@KaChen3000 I will update you on issue #615 once this PR has been merged and incorporated into a release.

File: source/slang/lower-to-ir.cpp [hunk 1]
```diff
@@ -3011,6 +3040,19 @@ LoweredValInfo tryGetAddress(
             // as part of the subscript operation being referenced.
             //
             auto subscriptInfo = val.getBoundSubscriptInfo();
+
+            // We don't want to immediately bind to a `ref` accessor if there is
+            // a `set` accessor available, unless we are in an "aggressive" mode
+            // where we really want/need a pointer to be able to make progress.
+            //
+            if(mode != TryGetAddressMode::Aggressive
+                && getMembersOfType<SetterDecl>(subscriptInfo->declRef).Count())
```
Comment (jsmall-nvidia) [inline source/slang/lower-to-ir.cpp]: Guess might be better to have a 'hasMembersOfType'

Comment (tfoleyNV) [reply source/slang/lower-to-ir.cpp]: Yes, that would be a good convenience function to provide.

**Pull request merged**